### DISCUSSION TITLE
How long were you stuck in a 'mockist' phase?

### DISCUSSION THREAD

**Discussion by ngrahovac** [techexcellenceio/techexcellence#40] [General]
Hi all!

As far as I can tell, many developers discovered TDD fairly recently, and were using a 'mockist' approach before.

How long were you stuck in this phase, and what was your light bulb moment that made you reconsider your approach?

**Comment by valentinajemuovic** [comment]
Great question!

I was stuck in it for many years, it was the only approach I knew, in fact all the tutorials I saw (like the Calculator example) were promoting testing classes and mocking out everything else.

The lightbulb moment was the pain of having to fix constantly breaking tests, the feeling that it's no longer "worth it", the complaints of others developers that unit testing is a waste of time due to breaking tests...

Another lightbulb moment was the realization that if I decide to refactor some code and replace it with design patterns (causes creation of new classes), I had to update all the tests.
